bot/exts/utils/thread_bumper.py

```diff
@@ -0,0 +1,142 @@
+
+from bot import constants
+from bot.bot import Bot
+from bot.log import get_logger
+from bot.pagination import LinePaginator
+from bot.utils import channel, scheduling
+
+log = get_logger(__name__)
+
+
+class ThreadBumper(commands.Cog):
+    """Cog that allow users to add the current thread to a list that get reopened on archive."""
+
+    # RedisCache[discord.Thread.id, "sentinel"]
+    threads_to_bump = RedisCache()
+
+    def __init__(self, bot: Bot):
+        self.bot = bot
+        self.init_task = scheduling.create_task(self.ensure_bumped_threads_are_active(), event_loop=self.bot.loop)
+
+    async def unarchive_threads_not_manually_archived(self, threads: list[discord.Thread]) -> None:
+        """
+        Iterate through and unarchive any threads that weren't manually archived recently.
+
+        This is done by extracting the manually archived threads from the audit log.
+
+        Only the last 200 thread_update logs are checked,
+        as this is assumed to be more than enough to cover bot downtime.
+        """
+        guild = self.bot.get_guild(constants.Guild.id)
+
+        recent_manually_archived_thread_ids = []
+        async for thread_update in guild.audit_logs(limit=200, action=discord.AuditLogAction.thread_update):
+            if getattr(thread_update.after, "archived", False):
+                recent_manually_archived_thread_ids.append(thread_update.target.id)
+
+        for thread in threads:
+            if thread.id in recent_manually_archived_thread_ids:
+                log.info(
+                    "#%s (%d) was manually archived. Leaving archived, and removing from bumped threads.",
+                    thread.name,
+                    thread.id
+                )
+                await self.threads_to_bump.delete(thread.id)
+            else:
+                await thread.edit(archived=False)
+
+    async def ensure_bumped_threads_are_active(self) -> None:
+        """Ensure bumped threads are active, since threads could have been archived while the bot was down."""
+        await self.bot.wait_until_guild_available()
+
+        threads_to_maybe_bump = []
+        for thread_id, _ in await self.threads_to_bump.items():
+            try:
+                thread = await channel.get_or_fetch_channel(thread_id)
+            except discord.NotFound:
+                log.info("Thread %d has been deleted, removing from bumped threads.", thread_id)
+                await self.threads_to_bump.delete(thread_id)
+                continue
+
+            if thread.archived:
+                threads_to_maybe_bump.append(thread)
+
+        await self.unarchive_threads_not_manually_archived(threads_to_maybe_bump)
+
+    @commands.group(name="bump")
+    async def thread_bump_group(self, ctx: commands.Context) -> None:
+        """A group of commands to manage the bumping of threads."""
+        if not ctx.invoked_subcommand:
+            await ctx.send_help(ctx.command)
+
+    @thread_bump_group.command(name="add")
```

Comment:
```suggestion
    @thread_bump_group.command(name="add", aliases=("a",))
```
How about adding `a` as an alias to `add`? Feel free to ignore this comment because `add` is short as it is, but, personally, I would expect `a` to be an alias for `add` since `r` is an alias for `remove`.

Reply:
https://github.com/python-discord/bot/pull/2084/commits/3346d71416fdb3223d0c4998f92e420886445fac